frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -237,7 +237,7 @@ export const getPipelineTasks = (
 
 export const containerToLogSourceStatus = (container: ContainerStatus): string => {
   if (!container) {
-    return LOG_SOURCE_WAITING;
+    return undefined;
   }
```

Comment:
This function's return type is string, we can add one more status eg: `LOG_SOURCE_NOT_READY`  and instead of returning undefined, we could return this status.